src/screens/signIn/index.js

```diff
@@ -1,56 +1,67 @@
+import React, {PureComponent} from 'react';
+import {View, Button} from '../../components';
+import {Input} from 'react-native-elements';
+import {signIn} from '../../store/actions';
+import {connect} from 'react-redux';
+import NavigationService from '../../services/NavigationService';
+import {styles} from './styles'
 
-import React, { PureComponent } from 'react';
-import { View,Text,Touchable } from '../../components';
-import {Button} from 'react-native-elements'
-import {Image, StyleSheet} from 'react-native'
-import * as COLORS from "../../../src/constants/colors"
-
+class SignIn extends PureComponent {
+  state = {
+    email: '',
+    password: '',
+  };
 
+  handleClick = () => {
+    this.props.onSignIn(this.state.email.trim(), this.state.password);
+  };
+  static getDerivedStateFromProps(props) {
+    if (props.loginIn != undefined) {
+      if (props.loginIn != false) NavigationService.navigate('Posts');
+    }
+    return null
+  }
 
-export default class SignIn extends PureComponent {
-    render() {
-      return (
-       
-        <View style={styles.container}>
-           <Image
-          style={styles.logo}
-          source={{uri : "https://upload.wikimedia.org/wikipedia/ru/thumb/9/9f/Twitter_bird_logo_2012.svg/1024px-Twitter_bird_logo_2012.svg.png"}}
+  render() {
+    return (
+      <View style={styles.container}>
+        <Input
+          style={styles.input}
+          onChangeText={email => this.setState({email})}
```

Comment:
Avoid using arrow function into render, in case of performance.
**ESPECIALLY** into PureComponents